src/main/java/org/kainos/ea/db/JobDao.java

```diff
@@ -43,9 +44,72 @@ public List<Job> getAllJobs(Connection connection) throws SQLException {
             allJobs.add(job);
 
         }
-
         return allJobs;
+    }
+
+    public Job getJobById(Connection connection, int id) throws SQLException {
+        String SQL = "SELECT job_id, job_title, job_spec_summary, capability.name AS capability, job_URL, job_band_name " +
+            "FROM job " +
+            "INNER JOIN job_band USING (job_band_id)" +
+            "INNER JOIN capability USING (capability_id)" +
+            "WHERE job_id = ?";
+
+
+        // Check database connection
+        if (connection == null) {
+            System.err.println("Failed to connect to database");
+            throw new SQLException("Failed to connect to database");
+        }
 
+
+        if(connection.isClosed()){
+            System.err.println("Database connection was closed");
+            throw new SQLException("Database connection was closed");
+        }
```

Comment:
this logic should be handles by the database connector

Reply:
Removed